### ISSUE TITLE
IsValidShopDomainAsync throws HttpRequestException when the shop name is invalid (not url)

### ISSUE THREAD

**Issue by datvm** [nozzlegear/ShopifySharp#202]
I think this function is supposed to handle all invalid case, including malfunctioned Url:

```
string urlFromUser = "foo";
bool isValidDomain = await AuthorizationService.IsValidShopDomainAsync(urlFromUser);
```

Current behavior: throwing HttpRequestException.
Expected: return `false`.

**Comment by nozzlegear** [comment]
Very interesting, thanks for reporting this! You're correct, the expected behavior is that this method returns false instead of throwing an exception. I'll get this fixed soon.

**Comment by datvm** [comment]
Hi I have forked the project and have fixed it. Didn't think you would reply so early. Will create Pull Request soon :)

**Comment by nozzlegear** [comment]
Great! I just happened to be looking at this repo when you created the issue :wink:

**Comment by datvm** [comment]
Wow that's fast. Thanks a lot! Just a small question: what is your iteration time to push changes to new release (to NuGet)? Just want to know when I can change my code to get from official NuGet instead of DLL file now :)

**Comment by nozzlegear** [comment]
It should be available in 4.8.3 within the next 45 minutes. As soon as the tests finish [on AppVeyor](https://ci.appveyor.com/project/nozzlegear/shopifysharp/build/4.0.122) I can publish to nuget and it takes about 20 minutes there before you can download it.

**Comment by nozzlegear** [comment]
Just published 4.8.3 to nuget, it'll show up for download in the next 20 minutes. Thanks for the pull request!

**Comment by kaikubadBS23** [comment]
i have build an app using this sdk and the app is live now.Suddenly i noticed that i am getting only 50 products.
var service = new ProductService(user.MyShopifyDomain, user.ShopifyAccessToken);
            int productCount = await service.CountAsync();
            IEnumerable<Product> product = await service.ListAsync();
this is the code i used to fetch all product.
please help
